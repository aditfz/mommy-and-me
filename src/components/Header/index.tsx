import React from 'react';
import './style/index.css';
function Header() {
  return (
    <div className='header'>
      <img
        className='header__logo'
        src='https://www.graphicsprings.com/filestorage/stencils/9ea51687148f4cfb53737a025d94caaa.png?width=500&height=500'
        alt='logo'
      />
      <h1 className='header__title'>Mommy & Me</h1>
    </div>
  );
}
export default Header;
