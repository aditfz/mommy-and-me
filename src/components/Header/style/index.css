.header {
  display: grid;
  grid-template-columns: 2fr 7fr;
  grid-column-gap: 20%;
  background-color: rgb(150, 243, 228);
}

.header__logo {
  width: 100%;
}

.header__title {
  font-family: 'Sigmar One', cursive;
  font-size: 6vw;
  color: rgb(252, 196, 11);
}
