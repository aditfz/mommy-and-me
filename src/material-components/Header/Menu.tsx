import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import useStyles from './styles/Menu.style';
const menuText = ['Home', 'Blog', 'Shop', 'Pictures', 'About Us', 'Contact Us'];
export default function HeaderMenu() {
  const classes = useStyles();
  let menue = menuText.map((item) => {
    return (
      <Grid key={item}  className={classes.menuItem}>
        <Typography variant='button' className={classes.menuItemText}>
          {item}
        </Typography>
      </Grid>
    );
  });
  return <Grid container> {menue}</Grid>;
}
