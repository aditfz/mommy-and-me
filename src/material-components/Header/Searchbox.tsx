import React, { useEffect, useRef } from 'react'
import { Paper, Grid } from '@material-ui/core'
import useStyles from './styles/SearchBox.style'
import SearchIcon from '@material-ui/icons/SearchRounded'

interface SearchboxProp {
  setSearchBoxIsOpen: (searchBoxIsOpen: boolean) => void
}
const SearchBox: React.FC<SearchboxProp> = ({setSearchBoxIsOpen}) => {
  const classes = useStyles()
  const searchBoxRef = useRef<HTMLDivElement>(null)

  // componentDidMount
  useEffect(() => {
    const myFunction = (event: MouseEvent) => {
      if (
        
        !searchBoxRef.current!.contains(event.target as Node)
      ) {
        setSearchBoxIsOpen(false)
      }
    }
    document.addEventListener('mousedown', myFunction)
    return () => {  
      document.removeEventListener('mousedown', myFunction)
    }
  }, [searchBoxRef, setSearchBoxIsOpen])

  return (
    <Paper className={classes.rootPaper} ref={searchBoxRef}>
      <Grid
        container
        justify='space-between'
        className={classes.searchBoxTextFielBox}
      >
        <input
          className={classes.searchTextField}
          placeholder='Search ...'
          autoFocus
        />
        <SearchIcon />
      </Grid>
    </Paper>
  )
}
export default SearchBox
