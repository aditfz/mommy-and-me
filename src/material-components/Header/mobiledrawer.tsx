import React, { useEffect, useRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import MailIcon from '@material-ui/icons/Mail';
import { NavLink } from 'react-router-dom';
const Logo = require('../../constants/images/logo.png');
const useStyles = makeStyles((theme) => ({
  list: {
    //   width: '100vw',
    //   maxWidth: 400,
    minHeight: 'fit-content',
    height: '100vh',
    width: 400,
    backgroundColor: '#2f2f2f',
    [theme.breakpoints.down('xs')]: {
      width: '100vw',
    },
  },
  fullList: {
    width: 'auto',
  },
  activeItem: {
    backgroundColor: 'black',
    color: 'white',
    borderRight: '2px solid red',
  },
}));
interface MobiledrawerProps {
  setDrawerIsOpen: (drawerIsOpen: boolean) => void;
}
const Mobiledrawer: React.FC<MobiledrawerProps> = ({ setDrawerIsOpen }) => {
  const drawerRef = useRef<HTMLDivElement>(null);

  const classes = useStyles();
  useEffect(() => {
    const myFunction = (event: MouseEvent) => {
      if (!drawerRef.current!.contains(event.target as Node)) {
        setDrawerIsOpen(false);
      }
    };
    document.addEventListener('mousedown', myFunction);
    return () => {
      document.removeEventListener('mousedown', myFunction);
    };
  }, [drawerRef]);

  return (
    <div
      ref={drawerRef}
      className={classes.list}
      role='presentation'
      //   onClick={toggleDrawer(anchor, false)}
      //   onKeyDown={toggleDrawer(anchor, false)}
    >
      <img src={Logo} alt='Logo' style={{ width: '60%' }} />
      <Divider />
      <List>
        {[
          'All mail',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'Spam',
          'Trash',
          'ttttttttt',
        ].map((text, index) => (
          <ListItem
            button
            key={text}
            component={NavLink}
            to={text}
            activeClassName={classes.activeItem}
          >
            <ListItemIcon>
              {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
            </ListItemIcon>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
    </div>
  );
};
export default Mobiledrawer;
