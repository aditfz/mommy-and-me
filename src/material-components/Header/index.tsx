import React, { useState } from 'react';
import { Paper, Grid, Hidden } from '@material-ui/core';
import useStyles from './styles/index.style';
import Menu from './Menu';
import SearchIcon from '@material-ui/icons/SearchRounded';
import SearchBox from './Searchbox';

export default function Header() {
  const classes = useStyles();

  const [searchBoxIsOpen, setSearchBoxIsOpen] = useState(false);

  return (
    <Paper className={classes.rootPaper}>
      <Grid container justify='space-between' alignItems='center'>
        <Grid item>
          <img
            className={classes.logo}
            src='https://www.graphicsprings.com/filestorage/stencils/9ea51687148f4cfb53737a025d94caaa.png?width=500&height=500'
            alt='logo'
          />
          <h1 style={{ fontFamily: 'Sigmar One', color: '#f00', margin: '0' }}>
            Mommy & Me
          </h1>
        </Grid>
        <Hidden smDown>
          <Grid item>
            <Menu />
          </Grid>
        </Hidden>
        <Grid item className={classes.searchBoxItem}>
          <div
            className={classes.searchIconBox}
            onClick={() => setSearchBoxIsOpen(!searchBoxIsOpen)}
          >
            <SearchIcon className={classes.searchIcon} />
          </div>
          {searchBoxIsOpen && (
            <SearchBox setSearchBoxIsOpen={setSearchBoxIsOpen} />
          )}
        </Grid>
      </Grid>
    </Paper>
  );
}
