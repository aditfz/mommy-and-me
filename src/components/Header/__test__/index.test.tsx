import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

test('smoke testing Header component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
});

// Snapshot testing Header component
test('Shallow rendering Header component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Header />);
  //   console.log(renderer.getRenderOutput());
  //   console.log(renderer.getRenderOutput().props.children);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
