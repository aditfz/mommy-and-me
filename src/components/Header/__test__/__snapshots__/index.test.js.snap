// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendering Header component 1`] = `
<div
  className="header"
>
  <img
    alt="logo"
    className="header__logo"
    src="https://www.graphicsprings.com/filestorage/stencils/9ea51687148f4cfb53737a025d94caaa.png?width=500&height=500"
  />
  <h1
    className="header__title"
  >
    Mommy & Me
  </h1>
</div>
`;
